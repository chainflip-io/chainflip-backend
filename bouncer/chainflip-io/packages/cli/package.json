{
  "name": "@chainflip-io/cli",
  "version": "0.0.25",
  "description": "",
  "main": "./dist/lib.js",
  "types": "./dist/lib.d.ts",
  "bin": {
    "chainflip": "./dist/cli.js"
  },
  "exports": {
    ".": {
      "import": "./dist/lib.mjs",
      "require": "./dist/lib.js",
      "types": "./dist/lib.d.ts"
    }
  },
  "typesVersions": {
    "*": {
      ".": [
        "dist/lib.d.ts"
      ]
    }
  },
  "files": [
    "dist",
    "README.md"
  ],
  "scripts": {
    "build": "rm -rf dist && tsup",
    "test": "jest",
    "typecheck": "tsc --noEmit"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "ethers": "5.7.2",
    "yargs": "^17.5.1",
    "zod": "^3.21.4"
  },
  "devDependencies": {
    "@types/yargs": "^17.0.24",
    "tsup": "^6.7.0",
    "tsx": "^3.12.6"
  },
  "repository": "https://github.com/chainflip-io/chainflip-sdk-monorepo.git",
  "publishConfig": {
    "registry": "https://npm.pkg.github.com"
  }
}
