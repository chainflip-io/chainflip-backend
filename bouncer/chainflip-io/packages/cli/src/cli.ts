import yargs from 'yargs/yargs';
import cliExecuteSwap, {
  yargsOptions as cliExecuteSwapOptions,
} from './commands/cliExecuteSwap';
import cliFundStateChainAccount, {
  yargsOptions as cliFundStateChainAccountOptions,
} from './commands/cliFundStateChainAccount';
import cliRequestSwapDepositAddress, {
  yargsOptions as cliRequestSwapDepositAddressOptions,
} from './commands/cliRequestSwapDepositAddress';

export default async function cli(args: string[]) {
  return yargs(args)
    .scriptName('chainflip-cli')
    .usage('$0 <cmd> [args]')
    .command('swap', '', cliExecuteSwapOptions, cliExecuteSwap)
    .command(
      'fund-state-chain-account',
      '',
      cliFundStateChainAccountOptions,
      cliFundStateChainAccount,
    )
    .command(
      'request-swap-deposit-address',
      '',
      cliRequestSwapDepositAddressOptions,
      cliRequestSwapDepositAddress,
    )
    .wrap(0)
    .strict()
    .help()
    .parse();
}
