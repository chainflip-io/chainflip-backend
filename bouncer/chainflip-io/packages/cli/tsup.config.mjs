/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig } from 'tsup';

export default defineConfig({
  treeshake: true,
  minify: false,
  dts: true,
  format: 'esm',
  entry: {
    lib: 'src/lib/index.ts',
    cli: 'src/main.ts',
  },
  sourcemap: true,
  target: 'es2022',
  banner: {
    js: `
import { fileURLToPath } from 'url';
import { createRequire as topLevelCreateRequire } from 'module';
import * as path from 'path';
const require = topLevelCreateRequire(import.meta.url);
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
    `,
  },
});
