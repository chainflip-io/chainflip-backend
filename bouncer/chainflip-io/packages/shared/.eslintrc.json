{
  "extends": "../../.eslintrc.json",
  "plugins": ["eslint-plugin-n"],
  "rules": {
    "n/no-process-env": ["error"]
  },
  "overrides": [
    {
      "files": ["*.test.ts", "*.mjs", "*.js", "*.cjs"],
      "rules": {
        "n/no-process-env": ["off"]
      }
    }
  ]
}
