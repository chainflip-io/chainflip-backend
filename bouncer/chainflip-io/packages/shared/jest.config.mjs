// eslint-disable-next-line import/no-relative-packages
import baseConfig from '../../jest.config.mjs';

process.env.RPC_BROKER_HTTPS_URL = 'wss://rpc-broker.example.com';

export default {
  ...baseConfig,
};
