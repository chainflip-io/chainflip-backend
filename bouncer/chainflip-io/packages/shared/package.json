{
  "name": "@chainflip-io/sdk-shared",
  "version": "1.0.0",
  "description": "",
  "scripts": {
    "typecheck": "tsc --noEmit",
    "clean": "rm -rf dist",
    "build": "tsc -p tsconfig.cjs.json && tsc -p tsconfig.mjs.json",
    "build:abis": "typechain --target=ethers-v5 --out-dir=./src/abis ./abis/**",
    "lint": "pnpm eslint --max-warnings 0 './**/*.ts'",
    "test": "jest",
    "test:ci": "CI=1 pnpm test -- --forceExit"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "peerDependencies": {
    "zod": "^3.21.4"
  },
  "dependencies": {
    "@polkadot/util": "^12.0.1",
    "@polkadot/util-crypto": "^12.0.1",
    "ethers": "5.7.2",
    "rxjs": "^7.8.0",
    "ws": "^8.13.0"
  },
  "devDependencies": {
    "@ethersproject/abi": "^5.7.0",
    "@ethersproject/providers": "^5.7.2",
    "@typechain/ethers-v5": "^11.0.0",
    "@types/ws": "^8.5.4",
    "typechain": "^8.2.0",
    "winston": "^3.8.2",
    "zod": "^3.21.4"
  }
}
