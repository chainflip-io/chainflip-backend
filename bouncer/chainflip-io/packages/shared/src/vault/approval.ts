import { ContractReceipt } from 'ethers';
import {
  checkAllowance,
  getTokenContractAddress,
  getVaultManagerContractAddress,
  approve,
} from '../contracts';
import { assert } from '../guards';
import { ExecuteOptions, TokenSwapParams } from './schemas';

export const checkVaultAllowance = (
  params: Pick<TokenSwapParams, 'srcAsset' | 'amount'>,
  opts: ExecuteOptions,
): ReturnType<typeof checkAllowance> => {
  const erc20Address =
    opts.network === 'localnet'
      ? opts.srcTokenContractAddress
      : getTokenContractAddress(params.srcAsset, opts.network);

  assert(erc20Address !== undefined, 'Missing ERC20 contract address');

  const vaultContractAddress =
    opts.network === 'localnet'
      ? opts.vaultContractAddress
      : getVaultManagerContractAddress(opts.network);

  return checkAllowance(
    params.amount,
    vaultContractAddress,
    erc20Address,
    opts.signer,
  );
};

export const approveVault = async (
  params: Pick<TokenSwapParams, 'srcAsset' | 'amount'>,
  opts: ExecuteOptions,
): Promise<ContractReceipt | null> => {
  const { isAllowable, erc20, allowance } = await checkVaultAllowance(
    params,
    opts,
  );

  if (isAllowable) return null;

  const vaultContractAddress =
    opts.network === 'localnet'
      ? opts.vaultContractAddress
      : getVaultManagerContractAddress(opts.network);

  return approve(
    params.amount,
    vaultContractAddress,
    erc20,
    allowance,
    opts.nonce,
  );
};
