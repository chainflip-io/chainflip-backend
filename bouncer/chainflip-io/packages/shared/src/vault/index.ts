export { default as executeSwap } from './executeSwap';
export { default as executeCall } from './executeCall';
export type {
  ExecuteSwapParams,
  ExecuteCallParams,
  ExecuteOptions,
} from './schemas';
export * from './approval';
