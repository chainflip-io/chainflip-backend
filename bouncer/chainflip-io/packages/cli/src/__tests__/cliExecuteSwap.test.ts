import { executeSwap } from '@/shared/vault';
import cli from '../cli';

jest.mock('ethers');
jest.mock('@/shared/vault', () => ({
  executeSwap: jest
    .fn()
    .mockResolvedValue({ status: 1, transactionHash: 'example-tx-hash' }),
}));

const localnet = `swap
  --wallet-private-key 0x2
  --chainflip-network localnet
  --dest-asset USDC
  --amount 1000000000
  --dest-address 0x0
  --src-token-contract-address 0x0
  --vault-contract-address 0x0
  --eth-network test`;

describe('cli', () => {
  it.each([localnet, localnet.replace('localnet', 'sisyphos')])(
    'calls the correct handler with the proper arguments',
    async (args) => {
      const logSpy = jest.spyOn(global.console, 'log').mockImplementation();
      await cli(args.split(/\s+/));

      expect(executeSwap).toHaveBeenCalledTimes(1);
      expect(jest.mocked(executeSwap).mock.calls).toMatchSnapshot();
      expect(logSpy).toHaveBeenCalledWith(
        expect.stringContaining('Transaction hash: example-tx-hash'),
      );
    },
  );
});
