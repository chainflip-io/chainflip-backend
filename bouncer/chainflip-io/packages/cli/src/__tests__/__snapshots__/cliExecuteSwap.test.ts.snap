// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli calls the correct handler with the proper arguments 1`] = `
[
  [
    {
      "amount": "1000000000",
      "destAddress": "0x0",
      "destAsset": "USDC",
      "destChain": "Ethereum",
      "srcAsset": undefined,
    },
    {
      "network": "localnet",
      "signer": undefined,
      "srcTokenContractAddress": "0x0",
      "vaultContractAddress": "0x0",
    },
  ],
]
`;

exports[`cli calls the correct handler with the proper arguments 2`] = `
[
  [
    {
      "amount": "1000000000",
      "destAddress": "0x0",
      "destAsset": "USDC",
      "destChain": "Ethereum",
      "srcAsset": undefined,
    },
    {
      "network": "sisyphos",
      "signer": undefined,
    },
  ],
]
`;
